segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageTests.java

```diff
@@ -2561,6 +2561,53 @@ private void testParallelReadRequestsOnSingleSegmentWithReentry(int numberOfRequ
         CompletableFuture.allOf(futures).join();
     }
 
+    /**
+     * Test concurrent writes to storage system segments by simulating concurrent writes.
+     *
+     * @throws Exception Throws exception in case of any error.
+     */
+    @Test
+    public void testSystemSegmentConcurrency() throws Exception {
+
+        SegmentRollingPolicy policy = new SegmentRollingPolicy(2); // Force rollover after every 2 byte.
+        @Cleanup
+        TestContext testContext = getTestContext(ChunkedSegmentStorageConfig.DEFAULT_CONFIG.toBuilder().indexBlockSize(3).build());
+        if (!(testContext.metadataStore instanceof InMemoryMetadataStore)) {
+            return;
+        }
+
+        // Simulate some writes to system segment, this should cause some new chunks being added.
+        val n = 4;
```

Comment:
I suppose `n` is number of writes? Let's name the variables with descriptive names so we don't guess this later on.